src/neptune/internal/backends/hosted_neptune_backend.py

```diff
@@ -1094,7 +1094,7 @@ def search_leaderboard_entries(
                 sort_by=sort_by,
                 ascending=ascending,
                 sort_by_column_type=sort_by_column_type,
-                progress_bar=progress_bar,
+                progress_bar=False if limit < default_step_size else progress_bar,
```

Comment:
If `limit` is smaller than the `default_step_size` then it means that we will not fetch even a "full" single page, so no need to construct a progress bar for that

Reply:
But as we're always fetching the total number of matching entries -> is it needed?